ext/simplexml/simplexml.c

```diff
@@ -400,7 +400,7 @@ static void change_node_zval(xmlNodePtr node, zval *value)
 			}
 			break;
 		default:
-			php_error_docref(NULL, E_WARNING, "It is not possible to assign complex types to nodes");
+			zend_type_error("It's not possible to assign a complex type to nodes, %s given", zend_zval_type_name(value));
```

Comment:
Error condition seems to be untested.

Reply:
Uh, I couldn't trigger this error case :/